modules/jpg/image_loader_libjpeg_turbo.cpp

```diff
@@ -77,6 +78,108 @@ Error jpeg_turbo_load_image_from_buffer(Image *p_image, const uint8_t *p_buffer,
 	return OK;
 }
 
+static int _get_exif_orientation(Ref<FileAccess> p_file) {
+	const uint64_t file_size = p_file->get_length();
+
+	// APP1 segment.
+	bool app1_found = false;
+	while (!p_file->eof_reached()) {
+		if (p_file->get_8() != 0xFF || p_file->get_8() != 0xE1) {
+			continue;
+		}
+		app1_found = true;
+		break;
+	}
+	if (!app1_found) {
+		return -1;
+	}
+
+	// 2 bytes - APP1 Segment length.
+	// 6 bytes - "Exif\0\0".
+	// 2 bytes - TIFF Header: Endianness.
+	// 2 bytes - TIFF Header: Magic 42.
+	// 4 bytes - TIFF Header: Offset to IFD0.
+	if (file_size - p_file->get_position() < 16) {
+		return -1;
+	}
+	p_file->seek(p_file->get_position() + 2); // Skip length.
+	if (memcmp(p_file->get_buffer(6).ptr(), "Exif\0\0", 6) != 0) {
+		return -1;
+	}
+	const uint64_t tiff_header_start = p_file->get_position();
+	switch (p_file->get_16()) {
+		case 0x4949: { // "II" - Little endian.
+			p_file->set_big_endian(false);
+		} break;
+		case 0x4D4D: { // "MM" - Big endian.
+			p_file->set_big_endian(true);
+		} break;
+		default: {
+			ERR_FAIL_V_MSG(-1, "Invalid EXIF endianness.");
+		} break;
+	}
+	ERR_FAIL_COND_V_MSG(p_file->get_16() != 0x002A, -1, "Invalid EXIF TIFF magic.");
+
+	const uint32_t offset = tiff_header_start + p_file->get_32();
+	if (file_size - offset < 2) {
+		return -1;
+	}
+	p_file->seek(offset);
+	const uint16_t entry_count = p_file->get_16();
+
+	for (unsigned int i = 0; i < entry_count; i++) {
+		// 2 bytes - Tag ID
+		// 2 bytes - Data type
+		// 4 bytes - Number of items
+		// 4 bytes - Value or value offset
+		if (file_size - p_file->get_position() < 12) {
+			return -1;
+		}
+		const uint16_t tag = p_file->get_16();
+		const uint16_t type = p_file->get_16();
+		const uint32_t count = p_file->get_32();
+
+		// Orientation tag & SHORT type.
+		if (tag != 0x0112 || type != 3 || count != 1) {
+			p_file->seek(p_file->get_position() + 4); // Skip value/offset.
+			continue;
+		}
+		int orientation = p_file->get_16(); // SHORT is left-justified within the 4 bytes.
+		ERR_FAIL_COND_V_MSG(orientation <= 0 || 8 < orientation, -1, "Invalid EXIF orientation value.");
```

Comment:
```suggestion
		ERR_FAIL_COND_V_MSG(orientation <= 0 || orientation > 8, -1, "Invalid EXIF orientation value.");
```
Far more readable and usual